src/bp/core/services/middleware/state-manager.ts

```diff
@@ -40,6 +40,7 @@ export class StateManager {
     state.session = (session && session.session_data) || { lastMessages: [] }
     state.temp = (session && session.temp_data) || {}
     state.bot = await this.kvs.get(event.botId, this.BOT_GLOBAL_KEY)
+    state.history = []
```

Comment:
I'd call it something like `__stacktrace` to make it more obvious what that is and also keep it private (especially since experimental)